rewrite-gradle/src/main/java/org/openrewrite/gradle/ChangeDependency.java

```diff
@@ -291,9 +291,15 @@ private J.MethodInvocation updateDependency(J.MethodInvocation m, ExecutionConte
                             }
                             if (original != updated) {
                                 String replacement = updated.toStringNotation();
```

Comment:
Is this solution overfit to the problem? What if the artifactId is a variable? Are we having to introduce the below logic because toStringNotation strips the variable?

Reply:
Just to chime in on the "overfit" part is that there is exusting precedent in many of other Gradle recipes to this respect. I agree that only expecting for only the version to be a template variable is a little assumptive, but it is the most prevalent form and very rarely do I see either the group or artifact being parameterized in real life builds.

To take this a step further, even using parameters in the way described no longer aligns with Gradle best practices and the advice there is to utilize a version catalog instead.